component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -274,87 +419,56 @@ public String getValidAudience(String tenantDomain) throws IdentityOAuth2Excepti
         return audience;
     }
 
-
     /**
-     * Error is logged and return false
-     * @param errorMessage
-     * @return
+     * To retreive the processed JWT claimset.
+     *
+     * @param signedJWT signedJWT
+     * @return JWT claim set
      * @throws IdentityOAuth2Exception
      */
-    private boolean handleException(String errorMessage, Exception e) throws IdentityOAuth2Exception {
-        log.error(errorMessage, e);
-        return false;
-    }
+    public ReadOnlyJWTClaimsSet getClaimSet(SignedJWT signedJWT) throws OAuthClientAuthnException {
 
-    /**
-     * Message is logged and returns false
-     * @param errorMessage
-     * @return
-     */
-    private boolean logAndReturnFalse(String errorMessage) {
-        if (log.isDebugEnabled()) {
-            log.debug(errorMessage);
-        }
-        return false;
-    }
-
-    /**
-     * Message is logged and returns true
-     * @param errorMessage
-     * @return
-     */
-    private boolean logAndReturnTrue(String errorMessage) {
-        if (log.isDebugEnabled()) {
-            log.debug(errorMessage);
-        }
-        return true;
-    }
-
-    /** Get the claimset from JWT
-     * @param signedJWT Signed JWT
-     * @return Claim set
-     */
-    public ReadOnlyJWTClaimsSet getClaimSet(SignedJWT signedJWT) throws IdentityOAuth2Exception {
         ReadOnlyJWTClaimsSet claimsSet;
+        String errorMessage;
+        if (signedJWT == null) {
+            errorMessage = "No Valid Assertion was found for " + Constants.OAUTH_JWT_BEARER_GRANT_TYPE;
```

Comment:
getClaimSet method is invoked from different contexts.
shouldn't the error say that, no valid JWT is found, rather than specifying the grant.

Reply:
In this context this method is invoked for this grant type right. So I believe specifying the grant here is fine.